### PR TITLE
msvc, refactor: Avoid some rare compiler warnings

### PR DESCRIPTION
**Pull request by hebasto**
This PR:
- removes 3 warnings from the global `DisableSpecificWarnings` list
- improves code hygiene

### CODE REVIEW

**Comment by sipsorcery** [comment]
Wouldn't this be better in two separate PR's? 

I realise the code changes are trivial but as far as I can tell they're not Windows specific. It wouldn't be the first time an innocuos cast has had a side effect. IMHO the warnings removal and code changes should be in separate PR's.

**Comment by hebasto** [comment]
> I can tell they're not Windows specific.

True.

**Comment by Zhaojun-Liu** [comment]
Hi @hebasto,
We update the bitcoin source recently, and encountered the waring C4858, and found that this issue (https://github.com/bitcoin/bitcoin/issues/26017) has been fixed in commit 31168dd, could this PR be merged as soon as possible? Thanks.

File: src/bech32.cpp [hunk 1]
```diff
@@ -241,7 +241,7 @@ constexpr std::array<uint32_t, 25> GenerateSyndromeConstants() {
     std::array<uint32_t, 25> SYNDROME_CONSTS{};
     for (int k = 1; k < 6; ++k) {
         for (int shift = 0; shift < 5; ++shift) {
-            int16_t b = GF1024_LOG.at(1 << shift);
+            int16_t b = GF1024_LOG.at(1ULL << shift);
```
Comment (MarcoFalke) [inline src/bech32.cpp:244]: It seems msvc violates the c++ standard by incorrectly promoting this to 64 bit

**Comment by Zhaojun-Liu** [comment]
Any updates of this PR?  @hebasto @MarcoFalke Thank you~

**Comment by hebasto** [comment]
> Wouldn't this be better in two separate PR's?
> 
> I realise the code changes are trivial but as far as I can tell they're not Windows specific. It wouldn't be the first time an innocuos cast has had a side effect. IMHO the warnings removal and code changes should be in separate PR's.

I separated some changes into #26189.

File: src/script/script.h [hunk 2]
```diff
@@ -209,7 +209,7 @@ enum opcodetype
 };
 
 // Maximum value that an opcode can be
-static const unsigned int MAX_OPCODE = OP_NOP10;
+static constexpr auto MAX_OPCODE = OP_NOP10;
```
[the remark this answers is not in the document]
Comment (hebasto) [reply src/script/script.h:212]: In addition, isn't clearer to move `MAX_OPCODE = OP_NOP10` into `enum opcodetype` definition?

File: src/bech32.cpp (hunk 1 above)
Comment (hebasto) [reply src/bech32.cpp:244]: The result of shift operation is promoted to `size_t` as [`std::array::at()`](https://en.cppreference.com/w/cpp/container/array/at) expects an argument of such a type.

**Comment by hebasto** [comment]
Changes in `src/bech32.cpp` has been split into #26252.

Touching consensus code for the sake of a MSVC warning does not look worthy, although making an `enum opcodetype` underlying type explicit is a good [idea](https://github.com/bitcoin/bitcoin/pull/25819#discussion_r986788445).

So closing this PR.

**Pull request closed without merging**